src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,198 @@
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update car " + car, e);
+        }
+        updateResultTable(car);
+        return car;
+    }
+
+    private void updateResultTable(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete car: " + car, e);
+        }
+        addNewRelationship(car);
+    }
+
+    private void addNewRelationship(Car car) {
+        String query = "INSERT INTO cars_drivers(driver_id, car_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(2, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                preparedStatement.setLong(1, driver.getId());
+                preparedStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert new relationship for car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id FROM cars JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "WHERE cars_drivers.driver_id = ? AND cars.is_deleted = FALSE;";
+        List<Long> indexes = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                indexes.add(resultSet.getObject("id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get all cars by driverId + " + driverId, e);
+        }
+        return indexes.stream()
```

Comment:
Don't forget to add drivers to the car

Reply:
I add drivers to car in get(Long id) method. I save all id and pass it to get(long id) method in stream